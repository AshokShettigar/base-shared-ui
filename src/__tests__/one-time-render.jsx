import { shallow } from 'enzyme';
import OneTimeRender from 'platform-common-ui/one-time-render';

describe('components/one-time-render-wrapper', function() {
    const localStorageMock = (() => {
        let store = {};

        return {
            getItem (key) {
                return store[key] || null;
            },
            setItem (key, value) {
                store[key] = value.toString();
            },
            clear () {
                store = {};
            }
        };
    })();
    Object.defineProperty(window, 'localStorage', {
        value: localStorageMock
    });
    it('should not render the component', function() {
        const wrapper = shallow(<OneTimeRender keyToCheck="key-To-Check" />);
        expect(wrapper).toMatchSnapshot();
    });
});
