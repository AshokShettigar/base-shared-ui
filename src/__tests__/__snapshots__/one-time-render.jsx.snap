// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/one-time-render-wrapper should not render the component 1`] = `
<div
  className="display-flex"
  data-component="OneTimeRenderWrapper"
/>
`;
